server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionWorker.java

```diff
@@ -46,25 +46,29 @@ public CompactionWorker(
 
   @Override
   public void run() {
-    while (!Thread.currentThread().isInterrupted()) {
```

Comment:
Two more easy solutions:
1. Let CompactionWorker extend WrappedRunnable which will automatically log the errors.
2. Use `taskExecutionPool.execute()` instead of `taskExecutionPool.submit()`, which will make use of `IoTDBDefaultThreadExceptionHandler` if any exception is thrown.